### PR TITLE
Fix compiler warnings

### LINKED ISSUE #4219: test build warnings: baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj
There should be no build warnings.

```
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(46): warning C4244: 'initializing': conversion from 'double' to 'float', possible loss of data [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(104): warning C4244: '=': conversion from 'double' to 'float', possible loss of data [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(160): warning C4244: '=': conversion from 'double' to 'float', possible loss of data [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(46): warning C4189: 'f': local variable is initialized but not referenced [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(103): warning C4101: 'x': unreferenced local variable [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
    24>d:\gh\coreclr\tests\src\baseservices\exceptions\regressions\dev11\147911\fpcw.cpp(159): warning C4101: 'x': unreferenced local variable [D:\gh\coreclr\bin\tests\obj\Windows_NT.x64.Debug\Native\src\baseservices\exceptions\regressions\Dev11\147911\fpcw.vcxproj]
```

### PR DESCRIPTION
**Pull request by rahku**
Fixes #4219

### CODE REVIEW

**Comment by rahku** [comment]
@dotnet-bot test Windows_NT x64 Release Priority 1 Build and Test

File: tests/src/baseservices/exceptions/regressions/Dev11/147911/fpcw.cpp [hunk 1]
```diff
@@ -101,12 +102,12 @@ extern "C" FPCW_API int RaiseFPException(void)
...
     __try
     {
-         double a = 1e-40, b;
-	 float  x, y;
-	y = a;
+        double a = 1e-40, b;
+        float  y;
+        y = (float)a;
         b = 2.0;
         printf("Shouldnt execute!\n");
```
Comment (jasonwilliams200OK) [inline tests/src/baseservices/exceptions/regressions/Dev11/147911/fpcw.cpp]: nit: s/`Shouldnt`/`Shouldn't`

Comment (rahku) [inline tests/src/baseservices/exceptions/regressions/Dev11/147911/fpcw.cpp]: thanks will take care of it.

**Comment by rahku** [comment]
PTAL @BruceForstall

**Comment by rahku** [comment]
FYI @RussKeldorph

**Comment by RussKeldorph** [comment]
LGTM

**Comment by rahku** [comment]
checks ran to completion before update to comments.

**Comment by BruceForstall** [comment]
LGTM

**Pull request merged**